Content.Server/Medical/CryoPodSystem.cs

```diff
@@ -0,0 +1,384 @@
+    [Dependency] private readonly SharedAppearanceSystem _appearanceSystem = default!;
+    [Dependency] private readonly MobStateSystem _mobStateSystem = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<CryoPodComponent, ComponentInit>(OnComponentInit);
+        SubscribeLocalEvent<CryoPodComponent, AtmosDeviceUpdateEvent>(OnCryoPodUpdateAtmosphere);
+        SubscribeLocalEvent<CryoPodComponent, GetVerbsEvent<AlternativeVerb>>(AddAlternativeVerbs);
+        SubscribeLocalEvent<CryoPodComponent, DragDropEvent>(HandleDragDropOn);
+        SubscribeLocalEvent<CryoPodComponent, InteractUsingEvent>(OnInteractUsing);
+        SubscribeLocalEvent<CryoPodComponent, DoInsertCryoPodEvent>(DoInsertCryoPod);
+        SubscribeLocalEvent<CryoPodComponent, ExaminedEvent>(OnExamined);
+        SubscribeLocalEvent<CryoPodComponent, PowerChangedEvent>(OnPowerChanged);
+        SubscribeLocalEvent<CryoPodComponent, GotEmaggedEvent>(OnEmagged);
+        SubscribeLocalEvent<CryoPodComponent, CryoPodPryFinished>(OnCryoPodPryFinished);
+        SubscribeLocalEvent<CryoPodComponent, CryoPodPryInterrupted>(OnCryoPodPryInterrupted);
+        SubscribeLocalEvent<CryoPodComponent, DestructionEventArgs>(OnDestroyed);
+        SubscribeLocalEvent<CryoPodComponent, GasAnalyzerScanEvent>(OnGasAnalyzed);
+        SubscribeLocalEvent<CryoPodComponent, ActivatableUIOpenAttemptEvent>(OnActivateUIAttempt);
+        SubscribeLocalEvent<CryoPodComponent, AfterActivatableUIOpenEvent>(OnActivateUI);
+    }
+
+    private void OnComponentInit(EntityUid uid, CryoPodComponent cryoPodComponent, ComponentInit args)
+    {
+        base.Initialize();
+        cryoPodComponent.BodyContainer = _containerSystem.EnsureContainer<ContainerSlot>(uid, $"scanner-bodyContainer");
+    }
+
+    private void UpdateAppearance(EntityUid uid, CryoPodComponent? cryoPod = null)
+    {
+        if (!Resolve(uid, ref cryoPod))
+            return;
+
+        var cryoPodEnabled = HasComp<ActiveCryoPodComponent>(uid);
+        _appearanceSystem.SetData(uid, SharedCryoPodComponent.CryoPodVisuals.ContainsEntity, cryoPod.BodyContainer.ContainedEntity == null);
+        _appearanceSystem.SetData(uid, SharedCryoPodComponent.CryoPodVisuals.IsOn, cryoPodEnabled);
+        if (TryComp<PointLightComponent>(uid, out var light))
+        {
+            light.Enabled = cryoPodEnabled && cryoPod.BodyContainer.ContainedEntity != null;
+        }
+
+        _appearanceSystem.SetData(uid,SharedCryoPodComponent.CryoPodVisuals.PanelOpen, false);
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+        foreach (var (_, cryoPod) in EntityQuery<ActiveCryoPodComponent, CryoPodComponent>())
+        {
+            cryoPod.Accumulator += frameTime;
+
+            if (cryoPod.Accumulator < cryoPod.BeakerTransferTime)
+                continue;
+
+            cryoPod.Accumulator -= cryoPod.BeakerTransferTime;
```

Comment:
This should be turned into a target timespan instead, to avoid the use of accumulators. You would need to the IGameTiming dependency, and then use it like this:

```suggestion
            if (_gameTiming.CurTime < cryoPod.TargetTime)
                continue;

            cryoPod.TargetTime = cryoPod.TargetTime + TimeSpan.FromSeconds(cryoPod.BeakerTransferTime);
```

Reply:
Annnnd this should prob also take into account time paused, so
if (_gameTiming.CurTime < cryoPod.TargetTime **+ time entity paused** ) continue;

Reply:
What exactly do you mean by time entity paused? I saw that _gameTiming has a Paused bool but I haven't seen it used in a way similar to what you said anywhere, do you have an example? Thanks in advance